xcb.c

```diff
@@ -150,6 +150,9 @@ xcb_window_t open_fullscreen_window(xcb_connection_t *conn, xcb_screen_t *scr, c
     values[0] = XCB_STACK_MODE_ABOVE;
     xcb_configure_window(conn, win, XCB_CONFIG_WINDOW_STACK_MODE, values);
 
+    /* Ensure that the window is created and set up before returning */
```

Comment:
Strictly speaking, this will only ensure that the request to create the window is sent to X11, not necessarily that the window is created before returning. For the latter, you’d need `xcb_aux_sync(conn);`, and I believe this is actually what we should be using here, if I understand the issue correctly.

I think the `fork()` call marks a point where either the parent or the child process can be scheduled first. The child process opens a separate X11 connection and expects the window which the parent process created in the parent’s X11 connection using `open_fullscreen_window` to be present. Without the parent using `xcb_aux_sync(conn);`, this is not necessarily true because the X server could process the client’s new connection while the parent’s connection hasn’t even flushed out the buffered requests yet.

If I captured this correctly, could you also add that to the commit message please? I think such an intricate bug deserves a detailed description :).